src/main/java/org/dita/dost/store/ant/types/StoreResource.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * This file is part of the DITA Open Toolkit project.
+ *
+ * Copyright 2020 Jarno Elovirta
+ *
+ * See the accompanying LICENSE file for applicable license.
+ */
+package org.dita.dost.store.ant.types;
+
+import org.apache.tools.ant.types.Resource;
+import org.dita.dost.util.Job;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URI;
+import java.util.Objects;
+
+public class StoreResource extends Resource {
+
+    private URI file;
+    private Job job;
+
+    public StoreResource(Job job, URI file) {
+        this.job = job;
+        this.file = file;
+    }
+
+    /**
+     * Get the name of this StoreResource.  If the basedir is set,
+     * the name will be relative to that.  Otherwise the basename
+     * only will be returned.
+     * @return the name of this resource.
+     */
+    @Override
+    public String getName() {
+        return file.getPath();
+    }
+
+    /**
+     * Learn whether this file exists.
+     * @return true if this resource exists.
+     */
+    @Override
+    public boolean isExists() {
```

Comment:
Another trivial comment that `isExists` feels like a weird because you're not checking if a file is exists, you're checking whether `fileExists` or `uriExists` or whatever?

Reply:
It overrides a method from Ant's `Resource` class, so we can't rename it to anything that makes more sense.